plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -521,12 +637,21 @@ void state_history_plugin::plugin_initialize(const variables_map& options) {
       if (auto resmon_plugin = app().find_plugin<resource_monitor_plugin>())
          resmon_plugin->monitor_directory(state_history_dir);
 
-      auto ip_port         = options.at("state-history-endpoint").as<string>();
-      auto port            = ip_port.substr(ip_port.find(':') + 1, ip_port.size());
-      auto host            = ip_port.substr(0, ip_port.find(':'));
-      my->endpoint_address = host;
-      my->endpoint_port    = std::stoi(port);
-      idump((ip_port)(host)(port));
+      auto ip_port = options.at("state-history-endpoint").as<string>();
+      if (ip_port.size()) {
+         auto port            = ip_port.substr(ip_port.find(':') + 1, ip_port.size());
+         auto host            = ip_port.substr(0, ip_port.find(':'));
+         my->endpoint_address = host;
+         my->endpoint_port    = std::stoi(port);
+         idump((ip_port)(host)(port));
```

Comment:
Remove debug dump, or convert to a proper `fc::dlog(_log, ...` call.

Reply:
Thank you, done, not sure about msg in dlog if we need any